src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -146,8 +147,15 @@ private async ValueTask InitializeAsync(PackageManagerModel model, INuGetUILogge
 
             _nugetPackageFileService?.Dispose();
             _nugetPackageFileService = await _serviceBroker.GetProxyAsync<INuGetPackageFileService>(NuGetServices.PackageFileService, CancellationToken.None);
+            var nuGetFeatureFlagService = await ServiceLocator.GetComponentModelServiceAsync<INuGetFeatureFlagService>();
+            var readmeTabEnabled = await nuGetFeatureFlagService.IsFeatureEnabledAsync(NuGetFeatureFlagConstants.RenderReadmeInPMUI);
+            if (readmeTabEnabled)
+            {
+                var editorOptionsFactoryService = await ServiceLocator.GetComponentModelServiceAsync<IEditorOptionsFactoryService>();
```

Comment:
This line will increase the number of times this method hangs as it will force types and assemblies to be faulted in on the UI thread. It is [already is hanging]((https://prism.vsdata.io/events/?eventType=UIDelay&query=c%3DPackageManagerControl.InitializeAsync), this will increase it.

Retrieve the `IEditorOptionsFactoryService` off the UI thread;

``` C#
await TaskScheduler.Default;

var editorOptionsFactoryService = await ServiceLocator.GetComponentModelServiceAsync<IEditorOptionsFactoryService>();

await SwitchToMainThreadAsync(cancellationToken);
```
Note you need a cancellation token, otherwise the comment I said yesterday will apply; it will crash VS on resume.